pkg/gardenlet/controller/federatedseed/network-policy/endpoints_controller.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network_policy
+
+import (
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/gardenlet/controller/federatedseed/network-policy/helper"
+
+	"k8s.io/client-go/tools/cache"
+)
+
+func (c *Controller) endpointAdd(_ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) endpointUpdate(_, _ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) endpointDelete(_ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) enqueueNamespaces() {
+	gardenNamespace, err := c.namespaceLister.List(GardenNamespaceSelector)
+	if err != nil {
+		c.log.Errorf("Failed to update Network Policy %q - unable to get the garden namespace from the lister: %v", helper.AllowToSeedAPIServer, err)
+		return
+	}
+	// expecting only one namespace that is labeled as Gardener namespace
+	if len(gardenNamespace) != 1 {
+		// happens sometimes on Gardenlet startup when the informer is not synced
+		// no problem, as the namespaces (including the garden ns) are enqueued individually
+		c.log.Debugf("Failed to update Network Policy %q - expected one namespace with labels %s:%s", helper.AllowToSeedAPIServer, v1beta1constants.LabelRole, v1beta1constants.GardenNamespace)
+		return
+	}
```

Comment:
Even after your answer I still don't understand this case. Why do we need it? If there is no `garden` namespace then there is nothing to be done - see above comment about Update/Delete handler for namespaces.

Reply:
Ah, and btw, why do you list in the first place? Can't you directly `GET` the namespace with name `garden`? I think that's a fair assumption that can be made?

Reply:
> Ah, and btw, why do you list in the first place? Can't you directly GET the namespace with name garden? I think that's a fair assumption that can be made?

I though having that label qualifies a namespace to be the "garden" namespace - and not the name "garden" itself. But if we make the assumption about the "garden" namespace already in other place, I can also just directly retrieve it.

> Even after your answer I still don't understand this case. Why do we need it? If there is no garden namespace then there is nothing to be done - see above comment about Update/Delete handler for namespaces.

Do you mean the endpoints controller should
- a) stop there and not update the network policies (like today)
- b) continue and update the Network policies for all Shoot namespaces (ignoring the missing garden namespace)

Reply:
Definitely b)